include/swift/AST/DiagnosticConsumer.h

```diff
@@ -178,8 +183,15 @@ class FileSpecificDiagnosticConsumer : public DiagnosticConsumer {
 
 private:
   void computeConsumersOrderedByRange(SourceManager &SM);
-  DiagnosticConsumer *consumerForLocation(SourceManager &SM,
-                                          SourceLoc loc) const;
+
+  /// Returns nullptr if diagnostic is to be suppressed,
+  /// None if diagnostic is to be distributed to every consumer,
+  /// a particular consumer if diagnostic goes there.
+  ///
+  /// (For Xcode compatibility, batch mode must suppress diagnostics to
+  /// non-primaries.)
```

Comment:
Did you mean "omitting diagnostics to non-primaries"? If emitting them is normal, then why are we *omitting* them if not for Xcode compatibility? What would be the accurate statement here?

Reply:
No, emitting is correct. We are *omitting* them so that we don't end up with a copy in multiple serialized diagnostic files. It's true that only Xcode reads those files right now, but that's not a "compatibility" issue, and it's not really part of the design here.

Reply:
Have rewritten that comment. Thanks for clarifying!